File: W3/homeWork_W3/Java/Basic/Complex/ComplexMain.java
public class ComplexMain {
    public static void main(String[] args) {
        Complex a = new Complex(2, 7);
        Complex b = new Complex(3, -8);
        
        // a.setX(2.0);
        // a.setY(7.0);
        
        System.out.println("a - " + a);
        System.out.println("a(x) - " + a.getReal());
        System.out.println("a(y) - " + a.getImaginary());
        System.out.println("b - " + b);
        System.out.println("b(x) - " + b.getReal());
        System.out.println("b(y) - " + b.getImaginary());
        
        a.increment(b);
        System.out.println("incremented a - " + a);
        System.out.println("a - " + a);
        System.out.println("b - " + b);
        
        a.decrement(b);
        System.out.println("Decremented a - " + a);
        System.out.println("a - " + a);
        System.out.println("b - " + b);
        
        Complex sum = a.add(b);
        System.out.println("Sum of a and b - " + sum);
        System.out.println("a - " + a);
        System.out.println("b - " + b);
        
        Complex diff = a.subtract(b);
        System.out.println("Diff of a and b - " + diff);
        System.out.println("a - " + a);
        System.out.println("b - " + b);
        
        Complex product = a.multiply(b);
        System.out.println("Product of a and b - " + product);
        System.out.println("a - " + a);
        System.out.println("b - " + b);
        
        if ( a.equals(b) ) {
            System.out.println("a is equal to b");    
        } else {
            System.out.println("a is not equal to b");
        }       
    }
}

File: PHP/Basic/Complex/ComplexMain.php
<?php
    require_once 'Complex.php';
    
    try {
        $a = new Complex(34.7, 31.8);
        $b = new Complex(12, 17);
        $c = new Complex(23, -28);
      
        echo "a.x: " . $a->real . ", a.y: " . $a->imaginary . PHP_EOL;
        echo "b: " . $b . PHP_EOL;
        echo "c: " . $c . PHP_EOL;
      
        $b->compare($c);
      
        $b->increment($c);
        $c->decrement($b);
        $sum = $b->add($c);
        $diff = $b->subtract($c);
        $product = $b->multiply($c);
        
        echo "Incremented b: " . $b . PHP_EOL;
        echo "Decremented c: " . $c . PHP_EOL;
        echo "Sum of the b and c - " . $sum . PHP_EOL;
        echo "Difference of the b and c - " . $diff . PHP_EOL;
        echo "Product of the b and c - " . $product . PHP_EOL;
        
    } catch( InvalidParam $ex ) {
        echo $ex->getMessage() . PHP_EOL;
    } catch ( InvalidAttribute $ex ) {
        echo $ex->getMessage() . PHP_EOL;
    }
?>
